test/Microsoft.Extensions.WebSockets.Test/WebSocketConnectionTests.ReceiveAsync.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Channels;
+using Microsoft.Extensions.WebSockets.Internal;
+using Xunit;
+
+namespace Microsoft.Extensions.WebSockets.Test
+{
+    public class WebSocketConnectionTests
+    {
+        public class TheReceiveAsyncMethod
+        {
+            [Theory]
+            [InlineData(new byte[] { 0x11, 0x00 }, WebSocketOpcode.Text, new byte[0])]
+            [InlineData(new byte[] { 0x11, 0x06, 0xA, 0xB, 0xC }, WebSocketOpcode.Text, new byte[] { 0xA, 0xB, 0xC })]
+            [InlineData(new byte[] { 0x11, 0x9, 0x1, 0x2, 0x3, 0x4, 0xA ^ 0x1, 0xB ^ 0x2, 0xC ^ 0x3, 0xD ^ 0x4 }, WebSocketOpcode.Text, new byte[] { 0xA, 0xB, 0xC, 0xD })]
+            public async Task ReadsSingleMessageFrames(byte[] rawFrame, WebSocketOpcode expectedOpcode, byte[] expectedPayload)
+            {
+                using (var factory = new ChannelFactory())
+                {
+                    var channel = factory.CreateChannel();
+
+                    var cts = new CancellationTokenSource();
+                    var cancellationToken = cts.Token;
+
+                    // Timeout for the test, but only if the debugger is not attached.
+                    if(!Debugger.IsAttached)
```

Comment:
It's really hard to debug the tests if they timeout while debugging, but when running the tests without the debugger we want them to time out if they are hanging because of some synchronization issue.